api/src/db-connection.ts

```diff
@@ -112,6 +112,31 @@ export class Connection {
     return isNaN(version) ? null : version;
   }
 
+  async getLatestProcessedArxivVersion(paperSelector: PaperSelector): Promise<number> {
+    if (isS2Selector(paperSelector)) {
+      // TODO (mjlangan): Error, or maybe null, think about this
+      return -1;
+    }
+    // Provided arXiv IDs might have a version suffix, but ignore that for this check.
+    const versionDelimiterIndex = paperSelector.arxiv_id.indexOf('v');
```

Comment:
Perhaps also take a look at Russell's code for stripping the version ID on the backend. One corner case I noticed in his code is that old-style arXiv IDs can have 'v' in them (like `math.VT/######v#`). I'm not sure if that's worth handling here, though maybe it is.

https://github.com/allenai/scholarphi/blob/aeb4d577d43f31f559dfd627f0ec2272d204006f/data-processing/common/commands/fetch_s2_data.py#L87

Reply:
Not sure I understand...are you saying that arxiv IDs are not guaranteed to be a series of digits followed by a `v#`?

Reply:
Ah, I see. The IDs prior to March 2007 throw a wrench in this.

Reply:
Going to punt on those older ones for now. @rreas says he doesn't plan on driving any papers that old through the pipeline yet. I think a better approach is for the pipeline to store this version as a separate column in the DB so the queries don't need to understand how to parse arxiv ID formats and can simply query integer values & sort them.